src/harness/testcases/WINNF_FT_S_HBT_testcase.py

```diff
@@ -0,0 +1,155 @@
+#    limitations under the License.
+import json
+import os
+import sas
+from  util import winnforum_testcase, makePpaAndPalRecordsConsistent
+import sas_testcase
+from datetime import datetime, timedelta
+import time
+
+
+class HeartbeatTestcase(sas_testcase.SasTestCase):
+  def setUp(self):
+    self._sas, self._sas_admin = sas.GetTestingSas()
+    self._sas_admin.Reset()
+
+  def tearDown(self):
+    pass
+
+  def getHeartBeatRequest(self,cbsd_ids, grant_ids, previousTransmitExpireTime):     
+    heartbeat_request = []
+    for cbsd_id, grant_id, transmitExpireTime in zip(cbsd_ids, grant_ids, previousTransmitExpireTime):
+        status = 'AUTHORIZED'
+        if(transmitExpireTime == None):
+            status = 'GRANTED'
+        elif datetime.strptime(transmitExpireTime,'%Y-%m-%dT%H:%M:%SZ') <=  (datetime.utcnow() + timedelta(seconds= 2)):
+            time.sleep(2)
+            status = 'GRANTED'
+        heartbeat_request.append({
+            'cbsdId': cbsd_id,
+            'grantId': grant_id,
+            'operationState': status
+        })
+    return {'heartbeatRequest': heartbeat_request}
+
+  def sendAndAssertSuccessfulHeartBeat(self, cbsd_ids, grant_ids, previousTransmitExpireTime):    
+    request =  self.getHeartBeatRequest(cbsd_ids, grant_ids, previousTransmitExpireTime)
+    response = self._sas.Heartbeat(request)['heartbeatResponse']
+    # check Heartbeat response 
+    self.assertEqual(len(response), 3)
+    for response_num, resp in enumerate(response):
+        self.assertEqual(resp['cbsdId'], cbsd_ids[response_num])
+        self.assertEqual(resp['grantId'], grant_ids[response_num])
+        self.assertEqual(resp['response']['responseCode'], 0)
+        grant_expire_time_1 = datetime.strptime(resp['grantExpireTime'],'%Y-%m-%dT%H:%M:%SZ')
+        transmit_expire_time_1 = datetime.strptime(resp['transmitExpireTime'],'%Y-%m-%dT%H:%M:%SZ')
+        self.assertLessEqual((transmit_expire_time_1 - datetime.utcnow()).total_seconds(), 240)
+        self.assertLessEqual(transmit_expire_time_1, grant_expire_time_1)
+    del request
+    return response
+   
+  @winnforum_testcase
+  def test_WINNF_FT_S_HBT_9(self):
+    """ Array request, of three cbsds,SAS terminates the Grant for the one
+    that impacts incumbent activity.
+
+    successful heartbeat for the grants of two cbsds that don't impact the incumbent
+    and failed with responseCode 500 for the third one.
+    """
+    # STEP 1
+    # register three devices 
+    device_c = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_c.json')))
+    device_d = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_d.json')))
+    device_e = json.load(
+        open(os.path.join('testcases', 'testdata', 'device_e.json')))
+    devices = [device_d, device_e, device_c]
+    for device in devices:
+        # inject FCC IDs
+        self._sas_admin.InjectFccId({'fccId': device['fccId']})
```

Comment:
Please add a comment to later inject user ID. The code to add the function to the admin interface is in a pending PR from Google; hopefully we can submit that soon so that you can update the code instead of leaving a "TODO".

Reply:
I modified the registration and grant requests, so I use assertRegisteredAndGranted